ga4gh/protocol.py

```diff
@@ -96,149 +175,28 @@ def isFull(self):
         The buffer is full if either (1) the number of items in the value
         list is >= pageSize or (2) the total length of the serialised
         elements in the page is >= maxResponseLength.
+
+        If page_size or max_response_length were not set in the request
+        then they're not checked.
         """
         return (
-            self._numElements >= self._pageSize or
-            self._valueListBuffer.tell() >= self._maxResponseLength)
+            (self._pageSize > 0 and self._numElements >= self._pageSize) or
+            (self._maxResponseLength > 0 and
```

Comment:
I've split it out and documented why. It's exactly what you noted - JSON conversion is a perf bottleneck, so don't bother checking if _maxResponseLength is set to the "don't care" value.